### PR TITLE
Raven db 8218

### PR DESCRIPTION
**Pull request by aviv86**

### CODE REVIEW

File: src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs [hunk 1]
```diff
@@ -1520,7 +1526,15 @@ private void VisitSelect(Expression operand)
                     {
                         var field = newExpression.Arguments[index] as MemberExpression;
                         if (field == null)
-                            continue;
+                        {
+                            // lambda 2 js
+                            TranslateSelectBodyToJs(newExpression);
```
Comment (ayende) [inline src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs:1531]: And what if this is nested?

File: src/Raven.Client/Documents/Session/IDocumentQuery.cs [hunk 2]
```diff
@@ -115,7 +115,8 @@ public interface IDocumentQuery<T> :
         /// <typeparam name="TProjection">Type of the projection.</typeparam>
         /// <param name="fields">Array of fields to load.</param>
         /// <param name="projections">Array of field projections.</param>
-        IDocumentQuery<TProjection> SelectFields<TProjection>(string[] fields, string[] projections);
+        /// <param name="fromAlias">From token alias.</param>
+        IDocumentQuery<TProjection> SelectFields<TProjection>(string[] fields, string[] projections, string fromAlias = null);
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/IDocumentQuery.cs]: Why do we need this `fromAlias` here?

File: src/Raven.Client/Util/JavascriptConversionExtensions.cs [hunk 3]
```diff
@@ -199,87 +225,109 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
...
-                if (type.GetGenericArguments().Length == 0)
-                    return false;
+                var nodeAsConst = context.Node as ConstantExpression;
 
-                return typeof(IEnumerable).IsAssignableFrom(type.GetGenericTypeDefinition());
+                if (nodeAsConst != null && nodeAsConst.Type == typeof(bool))
+                {
+                    context.PreventDefault();
+                    var writer = context.GetWriter();
+                    var val = nodeAsConst.Value.ToString().ToLower();
```
Comment (ayende) [inline src/Raven.Client/Util/JavascriptConversionExtensions.cs]: Use static values here.

File: src/Raven.Server/Documents/Patch/ScriptRunner.cs [hunk 4]
```diff
@@ -383,6 +386,22 @@ private JsValue LoadDocument(JsValue self, JsValue[] args)
...
+
+                var ticks = Convert.ToInt64(args[0].AsNumber());
+
+                var days = ticks / (24 * 60 * 60 * 1000);
+                var hours = (ticks / (60 * 60 * 1000) % 24);
+                var mins = (ticks / (60 * 1000) % 60);
+                var secs = (ticks /1000) % 60;
+                var ms = ticks % 1000;
+
+                return new JsValue($"{days}.{hours}:{mins}:{secs}.{ms}");
```
Comment (ayende) [inline src/Raven.Server/Documents/Patch/ScriptRunner.cs]: Use a timespan here instead of string.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Queries/QueryData.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+// <copyright file="QueryData.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Raven.Client.Documents.Queries
+{
+    public class QueryData
+    {
+        public string[] Fileds { get; set; }
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Queries/QueryData.cs:11]: typo: Fileds -> Fields

File: src/Raven.Client/Documents/Queries/QueryData.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
+//-----------------------------------------------------------------------
+// <copyright file="QueryData.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Raven.Client.Documents.Queries
+{
+    public class QueryData
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Queries/QueryData.cs:9]: I wonder if `QueryData` name isn't too wide for this class. Maybe `QuerySelectData`, `QuerySelection`

File: src/Raven.Client/Util/JavascriptConversionExtensions.cs [hunk 7]
```diff
@@ -199,87 +226,108 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
...
+
+                        writer.Write(")");
+
+                        switch (node.Member.Name)
+                        {
+                            case "Year":
+                                writer.Write(memberExpression.Member.Name == "UtcNow" ? ".getUTCFullYear()" : ".getFullYear()");
+                                break;
+                            case "Month":
+                                writer.Write(memberExpression.Member.Name == "UtcNow" ? ".getUTCMonth()+1" : ".getMonth()+1");
```
Comment (arekpalinski) [inline src/Raven.Client/Util/JavascriptConversionExtensions.cs:288]: `.getUTCMonth()+1` that's is tricky :)

**Pull request merged**